### PR TITLE
Upgrade efficientgo/{e2e|core} and e2e test dependencies

### PR DESCRIPTION
**Pull request by douglascamata**
These upgrades are due since a long time, there were plenty of releases, including breaking changes.

Assertions on metrics from `up` were changed from exact assertions to "great than or equal to". The reason is that it's difficult to know exactly how many queries/writes `up` will make due to its async nature and randomness introduced by resource usage on CI/CD containers/VMs. This made the tests flaky (often off by 1).

Also upgraded the Docker image tags from many e2e tests dependencies.

### CODE REVIEW

**Review by PhilipGough — DISMISSED** [review]
lgtm

File: test/e2e/tenants_test.go [hunk 1]
```diff
@@ -3,17 +3,18 @@
 package e2e
 
 import (
+	e2emon "github.com/efficientgo/e2e/monitoring"
```
Comment (squat) [inline test/e2e/tenants_test.go:6]: Nit: package ordering

**Review by squat — DISMISSED** [review]
Thanks @douglascamata this is tedious and helpful work! Lgtm, just two nits

**Comment by douglascamata** [comment]
@squat no problem and thanks for the review! I fixed the comments on https://github.com/observatorium/api/pull/434/commits/bdc7618f53cfe37d3997af6bc569d2e2813922f0.

**Review by squat — APPROVED** [review]

File: test/e2e/metrics_test.go [hunk 2]
```diff
@@ -5,14 +5,15 @@ package e2e
 import (
 	"context"
 	"fmt"
-	"github.com/efficientgo/e2e/matchers"
+	e2emon "github.com/efficientgo/e2e/monitoring"
+	"github.com/efficientgo/e2e/monitoring/matchers"
```
[the remark this answers is not in the document]
Comment (saswatamcode) [reply test/e2e/metrics_test.go]: Yeah this should be there in ci, raised PR for that: https://github.com/observatorium/api/pull/431

**Comment by saswatamcode** [comment]
Thanks!

**Pull request merged**